src/main/webapp/resources/js/pages/projects/share/ShareMetadata.jsx

```diff
@@ -0,0 +1,148 @@
+import { Table, Tag } from "antd";
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  getMetadataRestrictions,
+  useGetMetadataFieldsForProjectQuery,
+} from "../../../apis/metadata/field";
+import {
+  compareRestrictionLevels,
+  getColourForRestriction,
+} from "../../../utilities/restriction-utilities";
+import { setMetadataRestrictions } from "./shareSlice";
+import { TargetMetadataRestriction } from "./TargetMetadataRestriction";
+
+/**
+ * React component to display metadata restrictions.
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function ShareMetadata() {
+  const dispatch = useDispatch();
+  /**
+   * Available restrictions for metadata fields
+   */
+  const [restrictions, setRestrictions] = React.useState([]);
+
+  const { currentProject, projectId, metadataRestrictions } = useSelector(
+    (state) => state.shareReducer
+  );
+
+  /**
+   * Get the fields for the current project.  The restrictions from these fields
+   * will act as a base for the restriction level when the fields are shared.
+   */
+  const { data: sourceFields } = useGetMetadataFieldsForProjectQuery(
+    currentProject
+  );
+
+  /**
+   * Target project metadata fields. Needed to determine, which fields will be
+   * on both projects so that the new restriction for a field should start at
+   * the highest level of restriction.
+   */
+  const { data: targetExistingFields } = useGetMetadataFieldsForProjectQuery(
+    projectId,
+    {
+      skip: !projectId,
+    }
+  );
+
+  React.useEffect(() => {
+    if (sourceFields && targetExistingFields) {
+      const existing = targetExistingFields.reduce(
+        (fields, field) => ({
+          ...fields,
+          [field.fieldKey]: field,
+        }),
+        {}
+      );
+      const fields = sourceFields.map((field) => {
+        const f = { ...field, current: field.restriction };
```

Comment:
Change the variable `f` to something more descriptive

Reply:
Updated in 74eef49e57fa863cd71de04a70a14905a6af2f1c